devicetypes/vision-nameseca/zp3111jp-4-in-1-motion.src/zp3111jp-4-in-1-motion.groovy

```diff
@@ -0,0 +1,925 @@
+	}
+	return status
+}
+private getAttrStatusText(attrName, eventMaps=null) {
+	def status = (eventMaps?.find { it.name == attrName }?.descriptionText)
+	if (status) {
+		return status
+	}
+	else {
+		return getDescriptionText(device.currentState(attrName))
+	}
+}
+private getDescriptionText(data) {
+	switch (data?.name ?: "") {
+		case "motion":
+			return "${data.value}" == "active" ? "Motion" : "No Motion"
+			break
+		case "temperature":
+			return "${data.value}°${data.unit}"
+			break
+		case "humidity":
+			return  "${data.value}% RH"
+			break
+		case "lxLight":
+			return "${data.value} LUX"
+		case "pLight":
+			return "${data.value}% LIGHT"
+			break
+		default:
+			return ""
+	}
+}
+
+def refresh() {
+	log.debug "sending battery refresh command"
+	batteryGetCmd()
+}
+
+private createTamperEventMap(val) {
+	return createEventMap("tamper", val)
+}
+private createEventMaps(eventName, newVal, unit, displayed, onlyIfNew) {
+	def result = []
+
+	if (!onlyIfNew || getAttrValue(eventName) != newVal)
+	{
+		def eventMap = createEventMap(eventName, newVal, unit, displayed)
+		def desc = getDescriptionText(eventMap)
+
+		if (desc)
+		{
+			eventMap.descriptionText = desc
+		}
+		result << eventMap
+	}
+	return result
+}
+private createEventMap(eventName, newVal, unit="", displayed=null) {
+	def oldVal = getAttrValue(eventName)
+	def isNew = "${oldVal}" != "${newVal}"
+	def desc = "${eventName.capitalize()} is ${newVal}${unit}"
+
+	if (displayed == null) {
+		displayed = isNew
+	}
+
+	if (displayed) {
+		logDebug "${desc}"
+	}
+	else {
+		logTrace "${desc}"
+	}
+
+	return [
+			name: eventName,
+			value: newVal,
+			displayed: displayed,
+			isStateChange: true,
+			unit: unit,
+			descriptionText: "${device.displayName}: ${desc}"
+	]
+}
+private getAttrValue(attrName) {
+	try {
+		return device?.currentValue("${attrName}")
+	}
+	catch (ex) {
+		logTrace "$ex"
+		return null
+	}
+}
+private wakeUpIntervalGetCmd() {
+	return secureCmd(zwave.wakeUpV2.wakeUpIntervalGet())
+}
+private wakeUpIntervalSetCmd(val) {
+	return secureCmd(zwave.wakeUpV2.wakeUpIntervalSet(seconds:val, nodeid:zwaveHubNodeId))
+}
+private wakeUpNoMoreInfoCmd() {
+	return secureCmd(zwave.wakeUpV2.wakeUpNoMoreInformation())
+}
+private batteryGetCmd() {
+	return secureCmd(zwave.batteryV1.batteryGet())
+}
+private versionGetCmd() {
+	return secureCmd(zwave.versionV1.versionGet())
+}
+private sensorMultilevelGetCmd(sensorType) {
+	return secureCmd(zwave.sensorMultilevelV5.sensorMultilevelGet(scale: 2, sensorType: sensorType))
+}
+private configSetCmd(param, val) {
+	return secureCmd(zwave.configurationV2.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: val))
+}
+private configGetCmd(param) {
+	return secureCmd(zwave.configurationV2.configurationGet(parameterNumber: param.num))
+}
+private secureCmd(cmd) {
+	if (zwaveInfo?.zw?.contains("s") || ("0x98" in device.rawDescription?.split(" ")))
```

Comment:
I think you want just "98" and omit "0x"

Reply:
I have modified it to 98.
Thanks for your advice.